### PR TITLE
[LAP-40] Textures in OpenCL

### PR DESCRIPTION
**Pull request by Algo-ryth-mix**

### CODE REVIEW

**Comment by Algo-ryth-mix** [comment]
also references 
[LAP-48]
[LAP-40]

[LAP-48]: https://legion-engine.atlassian.net/browse/LAP-48
[LAP-40]: https://legion-engine.atlassian.net/browse/LAP-40

**Review by kukash — APPROVED** [review]

File: applications/sandbox/engine/shaderlib/texturemaps.shinc [hunk 1]
```diff
@@ -1,79 +0,0 @@
...
-
-vec4 LightingData(sampler2D MRDAoMap, vec2 uv)
-{
-	return texture(MRDAoMap, uv);
-}
-
-vec3 EmissiveMap(sampler2D emissiveMap, vec2 uv)
-{
-	return texture(emissiveMap, uv).rgb;
-}
```
Comment (GlynLeine) [inline applications/sandbox/engine/shaderlib/texturemaps.shinc]: pls no delete

File: legion/engine/core/compute/context.hpp [hunk 2]
```diff
@@ -65,6 +67,55 @@ class Context
...
+    {
+        size_type width = img.size.x;
+        size_type height = img.size.y;
+        size_type depth = 0;
+
+        cl_image_format fmt;
+
+        switch(img.format){
+        case channel_format::eight_bit: fmt.image_channel_data_type = CL_UNORM_INT8; break;
+        case channel_format::sixteen_bit: fmt.image_channel_data_type = CL_UNORM_INT8; break;
```
Comment (GlynLeine) [inline legion/engine/core/compute/context.hpp]: sixteen bit uint8?

Comment (Algo-ryth-mix) [reply legion/engine/core/compute/context.hpp]: CTRL-C,CTRL-V is a dangerous drug man. resolved in 7563ccc

File: applications/sandbox/engine/shaderlib/texturemaps.shinc (hunk 1 above)
Comment (Algo-ryth-mix) [reply applications/sandbox/engine/shaderlib/texturemaps.shinc]: weird merging artifact resolved in 2d940de

**Review by GlynLeine — APPROVED** [review]

**Pull request merged**